tests/recipes/cpc2/test_compute_haspi.py

```diff
@@ -0,0 +1,42 @@
+"""Tests for the haspi computation functions."""
+
+import numpy as np
+import pytest
+
+from recipes.cpc2.baseline.compute_haspi import (
+    parse_cec2_signal_name,
+    set_seed_with_string,
+)
+
+
+@pytest.mark.parametrize(
+    "signal_name, expected",
+    [("S1_L1_E1_hr", ("S1", "L1", "E1_hr")), ("S1_L1_E2", ("S1", "L1", "E2"))],
+)
+def test_parse_cec2_signal_name_ok(signal_name, expected):
+    """Test the parse_CEC2_signal_name function."""
+    assert parse_cec2_signal_name(signal_name) == expected
+
+
+@pytest.mark.parametrize(
+    "signal_name, expected",
+    [
+        ("S1", ValueError),
+        ("S1_L1", ValueError),
+        ("___", ValueError),
+        ("_X_X", ValueError),
+    ],
+)
+def test_parse_cec2_signal_name_error(signal_name, expected):
```

Comment:
I think this could be combined with the above `test_parse_cec2_signal_name_ok()` test to give one parameterised test function.

Untested but I think the following should work...


```
pytest.mark.parametrize(
    "signal_name, expected",
    [
         ("S1_L1_E1_hr", ("S1", "L1", "E1_hr")),
         ("S1_L1_E2", ("S1", "L1", "E2"))],
         ("S1", ValueError),
         ("S1_L1", ValueError),
         ("___", ValueError),
         ("_X_X", ValueError),
    ]
)
def test_parse_cec2_signal_name_ok(signal_name, expected):
    """Test the parse_CEC2_signal_name function."""
    with signal_name:
        assert parse_cec2_signal_name(signal_name) == expected
```

Not much reduction in code mind!